src/ConnectionAutomaton.cpp

```diff
@@ -322,10 +330,19 @@ void ConnectionAutomaton::handleUnknownStream(
 void ConnectionAutomaton::sendKeepalive() {
   Frame_KEEPALIVE pingFrame(
       FrameFlags_KEEPALIVE_RESPOND, folly::IOBuf::create(0));
-  connectionOutput_.onNext(pingFrame.serializeOut());
+  writeFrame(pingFrame.serializeOut());
 }
 
 void ConnectionAutomaton::onClose(ConnectionCloseListener listener) {
   closeListeners_.push_back(listener);
 }
+
+void ConnectionAutomaton::writeFrame(Payload frame) {
```

Comment:
nit: please consider naming this method outputFrame. It would be clearer to the me (reader) where is the frame going..